packages/@aws-cdk/aws-lambda-event-sources/lib/kinesis.ts

```diff
@@ -4,16 +4,24 @@ import * as cdk from '@aws-cdk/core';
 import { StreamEventSource, StreamEventSourceProps } from './stream';
 
 export interface KinesisEventSourceProps extends StreamEventSourceProps {
+  /**
+   * The time from which to start reading, in Unix time seconds.
+   *
+   * @default Required when startingPosition is AT_TIMESTAMP.
+   */
+  readonly startingPositionTimestamp?: number
 }
 
 /**
  * Use an Amazon Kinesis stream as an event source for AWS Lambda.
  */
 export class KinesisEventSource extends StreamEventSource {
+  private innerProps: KinesisEventSourceProps;
```

Comment:
I would prefer you not do this and just pull out the props you're going to use. Looks like just a `startingPositionTimestamp` is necessary.

Reply:
Done